doc/user/content/sql/create-source/kafka.md

```diff
@@ -0,0 +1,372 @@
+
+### SSL
+
+To connect to a Kafka broker that requires [SSL authentication](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html), use the provided [`WITH` options](#ssl-with-options).
+
+```sql
+CREATE MATERIALIZED SOURCE kafka_ssl
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'top-secret' WITH (
+      security_protocol = 'SSL',
+      ssl_key_location = '/secrets/materialized.key',
+      ssl_certificate_location = '/secrets/materialized.crt',
+      ssl_ca_location = '/secrets/ca.crt',
+      ssl_key_password = 'mzmzmz'
+  )
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'https://localhost:8081' WITH (
+      ssl_key_location = '/secrets/materialized.key',
+      ssl_certificate_location = '/secrets/materialized.crt',
+      ssl_ca_location = '/secrets/ca.crt',
+  );
+```
+
+Note that:
+
+- To connect to a Confluent Schema Registry using SSL, it must use the same certificate authority (CA) as the Kafka broker.
+
+#### SSL `WITH` options
+
+Field                      | Value  | Description
+---------------------------|--------|---------------------------
+`security_protocol`        | `text` | Use `ssl` to connect to the Kafka cluster.
+`ssl_certificate_location` | `text` | The absolute path to your SSL certificate. Required for SSL client authentication.
+`ssl_key_location`         | `text` | The absolute path to your SSL certificate's key. Required for SSL client authentication.
+`ssl_key_password`         | `text` | Your SSL key's password, if any.
+`ssl_ca_location`          | `text` | The absolute path to the certificate authority (CA) certificate. Used for both SSL client and server authentication. If unspecified, uses the system's default CA certificates.
+
+#### Confluent Schema Registry SSL `WITH` options
+
+Field | Value | Description
+------|-------|------------
+`ssl_certificate_location` | `text` | The absolute path to your SSL certificate. Required for SSL client authentication. <em>New in v0.17.0.</em>
+`ssl_key_location` | `text` | The absolute path to your SSL certificate's key. Required for SSL client authentication.  <em>New in v0.17.0.</em>
+`ssl_ca_location` | `text` | The absolute path to the certificate authority (CA) certificate. Used for both SSL client and server authentication. <em>New in v0.17.0.</em>
+
+### SASL
+
+To connect to a Kafka broker that requires [SASL authentication](https://docs.confluent.io/platform/current/kafka/authentication_sasl/auth-sasl-overview.html), use the provided [`WITH` options](#sasl-with-options).
+
+#### SASL/PLAIN
+
+```sql
+CREATE SOURCE kafka_sasl
+  FROM KAFKA BROKER 'broker.tld:9092' TOPIC 'top-secret' WITH (
+      security_protocol = 'SASL_SSL',
+      sasl_mechanisms = 'PLAIN',
+      sasl_username = '<BROKER_USERNAME>',
+      sasl_password = '<BROKER_PASSWORD>',
+  )
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'https://schema-registry.tld' WITH (
+      username = '<SCHEMA_REGISTRY_USERNAME>',
+      password = '<SCHEMA_REGISTRY_PASSWORD>'
```

Comment:
Maybe a preexisting issue, but I'm not sure the CSR `username`/`password` options are documented anywhere?

Reply:
@guswynn was looking into the basic HTTP authentication deets, not sure where these would fit. It ties a bit into your previous comment, hum. Maybe we need to call out the CSR as a separate section at some point...?

Reply:
#10150